ksql-cli/src/test/java/io/confluent/ksql/cli/console/JLineTerminalTest.java

```diff
@@ -78,4 +89,44 @@ public void shouldResetStatusMessage() {
     verify(statusBar)
         .update(ImmutableList.of(new AttributedString("", AttributedStyle.DEFAULT)));
   }
+
+  @Test
+  public void shouldWriteToSpool() throws IOException {
+    // Given:
+    final Writer spool = mock(Writer.class);
+    terminal.setSpool(spool);
+
+    // When:
+    terminal.writer().write(new char[]{'a'}, 0, 1);
+
+    // Then:
+    verify(spool).write(new char[]{'a'}, 0, 1);
+  }
+
+  @Test
+  public void shouldCloseSpoolOnUnset() throws IOException {
+    // Given:
+    final Writer spool = mock(Writer.class);
+    terminal.setSpool(spool);
+
+    // When:
+    terminal.unsetSpool();
+
+    // Then:
+    verify(spool).close();
+  }
+
+  @Test
+  public void shouldThrowOnTwoSpools() {
+    // Given:
+    final Writer spool = mock(Writer.class);
+    terminal.setSpool(spool);
+
+    // Expect:
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("Cannot set two spools!");
+
+    // When:
+    terminal.setSpool(spool);
+  }
 }
```

Comment:
Can we add a test showing that `readLine()` actually writes to the spool? Might also be nice to add a few simple tests for the new `Spool.java` and maybe also the `Console.java` changes, though I don't feel as strongly about those since they're mostly just piping.